File: File-Export-Service/src/main/java/com/kinok0/fileexportservice/entity/CommentEntity.java
package com.kinok0.fileexportservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Entity
@Table(name="comments")
@Getter
@Setter
@NoArgsConstructor
public class CommentEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "comment", nullable = false)
    private String comment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"user\"", referencedColumnName = "id")
    private UserEntity user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "task", referencedColumnName = "id")
    private TaskEntity task;

    public CommentEntity(String comment, UserEntity user, TaskEntity task) {
        this.comment = comment;
        this.user = user;
        this.task = task;
    }
}

File: Profile-Service/src/main/kotlin/com/kinok0/profileservice/entity/CommentEntity.kt
package com.kinok0.profileservice.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name="comments")
data class CommentEntity(
    @Id
    @GeneratedValue(strategy= GenerationType.UUID)
    var id: UUID? = null,

    @Column(name = "comment", nullable = false)
    var comment: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"user\"", referencedColumnName = "id")
    var user: UserEntity,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "task", referencedColumnName = "id")
    var task: TaskEntity
)
